prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/azure/AzureCreateVirtualMachineTask.java

```diff
@@ -0,0 +1,153 @@
+			}
+
+			AzureResourceManager azureResourceManager = AzureResourceManager.configure()
+					.withLogLevel(HttpLogDetailLevel.BASIC).authenticate(credential, profile).withDefaultSubscription();
+
+			log.info("Creating resource group...");
+			ResourceGroup resourceGroup = azureResourceManager.resourceGroups()
+					.define(resourcesPrefix + "ResourceGroup").withRegion(Region.US_EAST).create();
+
+			log.info("Creating availability set...");
+			AvailabilitySet availabilitySet = azureResourceManager.availabilitySets()
+					.define(resourcesPrefix + "AvailabilitySet").withRegion(Region.US_EAST)
+					.withExistingResourceGroup(resourcesPrefix + "ResourceGroup").create();
+
+			log.info("Creating public IP address...");
+			PublicIpAddress publicIPAddress = azureResourceManager.publicIpAddresses()
+					.define(resourcesPrefix + "PublicIP").withRegion(Region.US_EAST)
+					.withExistingResourceGroup(resourcesPrefix + "ResourceGroup").withDynamicIP().create();
+
+			log.info("Creating virtual network...");
+			Network network = azureResourceManager.networks().define(resourcesPrefix + "VN").withRegion(Region.US_EAST)
+					.withExistingResourceGroup(resourcesPrefix + "ResourceGroup").withAddressSpace("10.0.0.0/16")
+					.withSubnet(resourcesPrefix + "Subnet", "10.0.0.0/24").create();
+
+			log.info("Creating network interface...");
+			NetworkInterface networkInterface = azureResourceManager.networkInterfaces().define(resourcesPrefix + "NIC")
+					.withRegion(Region.US_EAST).withExistingResourceGroup(resourcesPrefix + "ResourceGroup")
+					.withExistingPrimaryNetwork(network).withSubnet(resourcesPrefix + "Subnet")
+					.withPrimaryPrivateIPAddressDynamic().withExistingPrimaryPublicIPAddress(publicIPAddress).create();
+
+			log.info("Creating virtual machine...");
+			VirtualMachine virtualMachine = azureResourceManager.virtualMachines().define(resourcesPrefix + "VM")
+					.withRegion(Region.US_EAST).withExistingResourceGroup(resourcesPrefix + "ResourceGroup")
+					.withExistingPrimaryNetworkInterface(networkInterface)
+					.withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_18_04_LTS)
+					.withRootUsername(userName).withSsh(sshKey).withComputerName(resourcesPrefix + "VM")
+					.withExistingAvailabilitySet(availabilitySet).withSize(VirtualMachineSizeTypes.STANDARD_D3_V2)
+					.create();
+
+			PublicIpAddress publicIpAddress = virtualMachine.getPrimaryPublicIPAddress();
+			if (publicIpAddress == null) {
+				log.error("VirtualMachine was created but without public IP");
+			}
+
+			String ipAddress = publicIpAddress.ipAddress();
+			log.info("VirtualMachine was created with public IP {}", ipAddress);
+			context.put("public-ip-address", ipAddress);
+
+			return new DefaultWorkReport(WorkStatus.COMPLETED, context, null);
+		}
+		catch (MissingParameterException e) {
+			log.error("Task {} failed: missing required parameter, error: {}", getName(), e.getMessage());
+		}
+		catch (Exception e) {
```

Comment:
please be specific which exceptions we want to catch here

Reply:
Except to `MissingParameterException` there is no known declared Exception that should be catched
Therefore I catch `Exception` to cover any unexpected `RuntimeException`